git diff 17c5e0e8014bd89da84bb00f1b60fb25e9fb1941..e339940cfbb7863b3bb4d8df22f40b29484fa32b -- ./libs/glib2/
diff --git a/libs/glib2/Makefile ../../../openwrt_trunk/feeds/packages/libs/glib2/Makefile
index b5a5667..21a7426 100644
--- a/libs/glib2/Makefile
+++ ../../../openwrt_trunk/feeds/packages/libs/glib2/Makefile
@@ -8,16 +8,17 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=glib2
-PKG_VERSION:=2.37.7
-PKG_RELEASE:=1
+PKG_VERSION:=2.26.1
+PKG_RELEASE:=3
 
-PKG_SOURCE:=glib-$(PKG_VERSION).tar.xz
+PKG_SOURCE:=glib-$(PKG_VERSION).tar.bz2
 PKG_BUILD_DIR:=$(BUILD_DIR)/glib-$(PKG_VERSION)
-PKG_SOURCE_URL:=@GNOME/glib/2.37
-PKG_MD5SUM:=17a9493e3e2efebc8ab81ec8d4480bfb
+PKG_SOURCE_URL:=@GNOME/glib/2.26
+PKG_MD5SUM:=17535accceef55bcb17a74d73f9c2aef
 
-PKG_BUILD_DEPENDS:=glib2/host libpthread zlib libintl libffi
-HOST_BUILD_DEPENDS:=libintl/host libiconv/host libffi/host
+PKG_FIXUP:=autoreconf
+PKG_BUILD_DEPENDS:=glib2/host libpthread zlib libintl
+HOST_BUILD_DEPENDS:=gettext/host libiconv/host
 PKG_INSTALL:=1
 
 HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/glib-$(PKG_VERSION)
@@ -29,7 +30,7 @@ include $(INCLUDE_DIR)/nls.mk
 define Package/glib2
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:=$(ICONV_DEPENDS) $(INTL_DEPENDS) +zlib +libpthread +libffi
+  DEPENDS:=$(ICONV_DEPENDS) $(INTL_DEPENDS) +zlib +libpthread
   TITLE:=glib 2.0
   URL:=http://www.gtk.org/
 endef
diff --git a/libs/glib2/patches/001-automake-compat.patch ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/001-automake-compat.patch
new file mode 100644
index 0000000..aa6833a
--- /dev/null
+++ ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/001-automake-compat.patch
@@ -0,0 +1,12 @@
+--- a/configure.ac
++++ b/configure.ac
+@@ -2741,7 +2741,8 @@ dnl **************************
+ dnl *** Checks for gtk-doc ***
+ dnl **************************
+ 
+-GTK_DOC_CHECK([1.15])
++AM_CONDITIONAL([ENABLE_GTK_DOC], [false])
++AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [false])
+ 
+ AC_ARG_ENABLE(man,
+               [AC_HELP_STRING([--enable-man],
diff --git a/libs/glib2/patches/002-missing-gthread-include.patch ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/002-missing-gthread-include.patch
new file mode 100644
index 0000000..8def409
--- /dev/null
+++ ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/002-missing-gthread-include.patch
@@ -0,0 +1,10 @@
+--- a/glib/gatomic.c
++++ b/glib/gatomic.c
+@@ -881,6 +881,7 @@ g_atomic_pointer_compare_and_exchange (v
+ #endif /* DEFINE_WITH_WIN32_INTERLOCKED */
+ 
+ #ifdef DEFINE_WITH_MUTEXES
++#include "gthread.h"
+ /* We have to use the slow, but safe locking method */
+ static GMutex *g_atomic_mutex; 
+ 
diff --git a/libs/glib2/patches/010-move-iconv-to-libs.patch ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/010-move-iconv-to-libs.patch
new file mode 100644
index 0000000..89b3fea
--- /dev/null
+++ ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/010-move-iconv-to-libs.patch
@@ -0,0 +1,10 @@
+--- a/glib-2.0.pc.in
++++ b/glib-2.0.pc.in
+@@ -10,6 +10,5 @@ glib_mkenums=glib-mkenums
+ Name: GLib
+ Description: C Utility Library
+ Version: @VERSION@
+-Libs: -L${libdir} -lglib-2.0 @INTLLIBS@
+-Libs.private: @ICONV_LIBS@
++Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@
+ Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
diff --git a/libs/glib2/patches/020-no-docs-tests-translations.patch ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/020-no-docs-tests-translations.patch
new file mode 100644
index 0000000..0460640
--- /dev/null
+++ ../../../openwrt_trunk/feeds/packages/libs/glib2/patches/020-no-docs-tests-translations.patch
@@ -0,0 +1,81 @@
+--- a/configure.ac
++++ b/configure.ac
+@@ -3737,18 +3737,6 @@ gio/fen/Makefile
+ gio/fam/Makefile
+ gio/win32/Makefile
+ gio/tests/Makefile
+-po/Makefile.in
+-docs/Makefile
+-docs/reference/Makefile
+-docs/reference/glib/Makefile
+-docs/reference/glib/version.xml
+-docs/reference/gobject/Makefile
+-docs/reference/gobject/version.xml
+-docs/reference/gio/Makefile
+-docs/reference/gio/version.xml
+-tests/Makefile
+-tests/gobject/Makefile
+-tests/refcount/Makefile
+ m4macros/Makefile
+ ])
+ 
+--- a/gio/Makefile.am
++++ b/gio/Makefile.am
+@@ -264,8 +264,6 @@ giowin32include_HEADERS = \
+ 
+ endif
+ 
+-SUBDIRS += tests
+-
+ libgio_2_0_la_SOURCES =		\
+ 	gappinfo.c 		\
+ 	gasynchelper.c 		\
+--- a/glib/Makefile.am
++++ b/glib/Makefile.am
+@@ -48,9 +48,9 @@ else
+ gatomic_c = gatomic.c
+ endif
+ 
+-SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre . tests
++SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre .
+ 
+-DIST_SUBDIRS = libcharset gnulib pcre update-pcre tests
++DIST_SUBDIRS = libcharset gnulib pcre update-pcre
+ 
+ AM_CPPFLAGS = 				\
+ 	$(glib_INCLUDES) 		\
+--- a/gobject/Makefile.am
++++ b/gobject/Makefile.am
+@@ -4,7 +4,7 @@
+ ## Process this file with automake to produce Makefile.in
+ include $(top_srcdir)/Makefile.decl
+ 
+-SUBDIRS = . tests
++SUBDIRS = .
+ 
+ CLEANFILES=
+ 
+--- a/gthread/Makefile.am
++++ b/gthread/Makefile.am
+@@ -1,8 +1,8 @@
+ ## Process this file with automake to produce Makefile.in
+ include $(top_srcdir)/Makefile.decl
+ 
+-SUBDIRS = . tests
+-DIST_SUBDIRS = tests
++SUBDIRS = .
++DIST_SUBDIRS = 
+ 
+ AM_CPPFLAGS = 				\
+ 	$(glib_INCLUDES) 		\
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -3,7 +3,7 @@ include $(top_srcdir)/Makefile.decl
+ 
+ ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
+ 
+-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
++SUBDIRS = . m4macros glib gmodule gthread gobject gio
+ DIST_SUBDIRS = $(SUBDIRS) build
+ 
+ bin_SCRIPTS = glib-gettextize
